Drop-In/src/main/java/com/braintreepayments/api/dropin/AddCardActivity.java

```diff
@@ -283,19 +286,26 @@ protected void createCard() {
                     .postalCode(cardForm.getPostalCode())
                     .validate(shouldVault);
 
-            if (shouldRequestThreeDSecureVerification()) {
-                ThreeDSecure.performVerification(mBraintreeFragment, cardBuilder,
-                        mDropInRequest.getAmount());
-            } else {
-                Card.tokenize(mBraintreeFragment, cardBuilder);
-            }
+            Card.tokenize(mBraintreeFragment, cardBuilder);
         }
     }
 
     @Override
     public void onPaymentMethodNonceCreated(PaymentMethodNonce paymentMethod) {
-        mBraintreeFragment.sendAnalyticsEvent("sdk.exit.success");
-        finish(paymentMethod, null);
+        if (shouldRequestThreeDSecureVerification() && mDropInRequest.getThreeDSecureRequest() == null) {
+            ThreeDSecureRequest threeDSecureRequest = new ThreeDSecureRequest().amount(mDropInRequest.getAmount());
+            mDropInRequest.threeDSecureRequest(threeDSecureRequest);
```

Comment:
The documentation for `threeDSecureRequest` on `DropInRequest` says that if the merchant doesn't provide an amount, we'll default to the amount on `DropInRequest`. But here, we're only defaulting the amount if the merchant doesn't provide a `threeDSecureRequest` (i.e., the request object is null). If the `threeDSecureRequest` is present, but the _amount_ is null, we don't provide a default.

@scannillo and I want to propose this alternative:
```
    @Override
    public void onPaymentMethodNonceCreated(PaymentMethodNonce paymentMethod) {
        if (!mRequestedThreeDSecure && shouldRequestThreeDSecureVerification()) {
            if (mDropInRequest.getThreeDSecureRequest() == null) {
                mDropInRequest.threeDSecureRequest(new ThreeDSecureRequest());
            }

            if (mDropInRequest.getThreeDSecureRequest().getAmount() == null) {
                mDropInRequest.getThreeDSecureRequest().amount(mDropInRequest.getAmount());
            }

            mDropInRequest.getThreeDSecureRequest().nonce(paymentMethod.getNonce());
            ThreeDSecure.performVerification(mBraintreeFragment, mDropInRequest.getThreeDSecureRequest());
            mRequestedThreeDSecure = true;
        } else {
            mBraintreeFragment.sendAnalyticsEvent("sdk.exit.success");
            finish(paymentMethod, null);
        }
  }
```